### PR TITLE
chore(templates): add comments to templates to describe resources

### PR DESCRIPTION
**Pull request by efekarakus**
These comments are parsed by the progress tracker to display
human-readable text to users explaining what each resource is.

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

**Review by iamhopaul123 — APPROVED** [review]
:shipit:

File: templates/task/cf.yml [hunk 1]
```diff
@@ -59,6 +60,7 @@ Resources:
       TaskRoleArn:
         !If [HasTaskRole, !Ref TaskRole, !Ref "AWS::NoValue"]
   DefaultExecutionRole:
+    # An IAM role to grant the Fargate agent to make AWS API calls on your behalf.
```
Comment (huanjani) [inline templates/task/cf.yml]: nit: can this wording match that of ExecutionRole yaml (below)?

File: templates/environment/versions/cf-v1.1.0.yml [hunk 2]
```diff
@@ -83,6 +85,7 @@ Resources:
 
   # Only accept requests coming from the public ALB or other containers in the same security group.
   EnvironmentSecurityGroup:
+    # A security group for your containers to talk to each other.
```
Comment (bvtujo) [inline templates/environment/versions/cf-v1.1.0.yml]:
```suggestion
    # A security group to allow your containers to talk to each other.
```

File: templates/task/cf.yml (hunk 1 above)
Comment (bvtujo) [reply templates/task/cf.yml]:
```suggestion
    # An IAM role to control permissions for the containers in your tasks.
```

**Review by huanjani — APPROVED** [review]
Looks great! Remove label whenever you're ready!

File: templates/workloads/common/cf/state-machine.yml [hunk 3]
```diff
@@ -1,4 +1,5 @@
 StateMachine:
+  # A state machine to invoke and handle failures for your job.
```
Comment (bvtujo) [inline templates/workloads/common/cf/state-machine.yml]:
```suggestion
  # A state machine to invoke your job and handle retry and timeout logic.
```

File: templates/workloads/common/cf/taskrole.yml [hunk 4]
```diff
@@ -1,4 +1,5 @@
 TaskRole:
+  # An IAM role to control permission for the containers in your tasks.
```
Comment (bvtujo) [inline templates/workloads/common/cf/taskrole.yml]:
```suggestion
  # An IAM role to control permissions for the containers in your tasks.
```

**Review by bvtujo — APPROVED** [review]
Nice, this is clever. Just a few nits.

**Pull request merged**